File: heifconverter/src/main/java/linc/com/heifconverter/iso14496/part12/PrimaryItemBox.java
package linc.com.heifconverter.iso14496.part12;

import org.mp4parser.support.AbstractFullBox;
import org.mp4parser.tools.IsoTypeReader;

import java.nio.ByteBuffer;

public final class PrimaryItemBox extends AbstractFullBox {
    public static final String TYPE = "pitm";

    private int itemId;

    public PrimaryItemBox() {
        super(TYPE);
    }

    public int getItemId() {
        return itemId;
    }

    @Override
    public void _parseDetails(ByteBuffer content) {
        parseVersionAndFlags(content);
        itemId = IsoTypeReader.readUInt16(content);
    }

    @Override
    public long getContentSize() {
        return 10;
    }

    @Override
    public void getContent(ByteBuffer byteBuffer) {
        throw new RuntimeException(TYPE + " not implemented");
    }

    @Override
    public String toString() {
        return "PrimaryItemBox[itemId=" + itemId + "]";
    }
}

File: heifconverter/src/main/java/linc/com/heifconverter/iso14496/part12/PrimaryItemBox.kt
package linc.com.heifconverter.iso14496.part12

import org.mp4parser.support.AbstractFullBox
import org.mp4parser.tools.IsoTypeReader
import java.nio.ByteBuffer

internal class PrimaryItemBox : AbstractFullBox(TYPE) {
    var itemId = 0
        private set

    public override fun _parseDetails(content: ByteBuffer) {
        parseVersionAndFlags(content)
        itemId = IsoTypeReader.readUInt16(content)
    }

    public override fun getContentSize(): Long {
        return 10
    }

    public override fun getContent(byteBuffer: ByteBuffer) {
        throw RuntimeException("$TYPE not implemented")
    }

    override fun toString(): String {
        return "PrimaryItemBox[itemId=$itemId]"
    }

    companion object {
        const val TYPE = "pitm"
    }
}